### PR TITLE
Add logging to GetLruPages

### PR DESCRIPTION
**Pull request by JuanCarlosGI**
Per call to GetLruPages, log the count of contentHashesWithInfo, and the age (not effective age) of the first and last element.

Per page, log the age and effective age of the first and last element of the page, and of the top element of the priority queue.

### CODE REVIEW

File: Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs [hunk 1]
```diff
@@ -249,6 +250,8 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
         {
             Contract.Assert(_configuration.HasReadOrWriteMode(ContentLocationMode.LocalLocationStore), "GetLruPages can only be called when local location store is enabled");
 
+            context.Debug($"GetLruPages start with contentHashesWithInfo.Count={contentHashesWithInfo.Count}, firstAge={contentHashesWithInfo.FirstOrDefault().Age}, lastAge={contentHashesWithInfo.LastOrDefault().Age}");
```
Comment (ljcollins25) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: >LastOrDefault [](start = 196, length = 13)

LastOrDefault is not optimized for IReadOnlyList<T>. Maybe use contentHashesWithInfo[contentHashesWithInfo.Count - 1] instead. Maybe handle the empty collection separately.

File: Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs [hunk 2]
```diff
@@ -275,6 +278,8 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
                     priorityQueue.Push(lastAccessTimes[i]);
                 }
 
+                context.Debug($"GetLruPages page.First:(Age={page.First().Age}, EffectiveAge={page.First().EffectiveAge}), queue.Top:(Age={priorityQueue.Top.Age}, EffectiveAge={priorityQueue.Top.EffectiveAge})");
```
Comment (ljcollins25) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: >First [](start = 49, length = 5)

Can we print the info the last item of the page as well? Maybe include hash of first and last item of page as well.

File: Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs [hunk 3]
```diff
@@ -275,6 +285,10 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
                     priorityQueue.Push(lastAccessTimes[i]);
                 }
 
+                first = page[0];
```
Comment (dannyvv) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: Guaranteed to have at least one page? If so perhaps comment...

File: Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs [hunk 4]
```diff
@@ -249,6 +249,16 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
...
 
+            ContentHashWithLastAccessTimeAndReplicaCount first = default;
+            ContentHashWithLastAccessTimeAndReplicaCount last = default;
+            if (contentHashesWithInfo.Count != 0)
+            {
+                first = contentHashesWithInfo[0];
+                last = contentHashesWithInfo[contentHashesWithInfo.Count - 1];
+            }
+
+            context.Debug($"GetLruPages start with contentHashesWithInfo.Count={contentHashesWithInfo.Count}, firstAge={first.Age}, lastAge={last.Age}");
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: Do you want to put this trace inside the if block maybe?

File: Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs [hunk 5]
```diff
@@ -275,6 +285,11 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
                     priorityQueue.Push(lastAccessTimes[i]);
                 }
 
+                //  Each page is guaranteed to have at least one entry.
+                first = page[0];
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: Contract.Assert(page.Count != 0)?

Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: nit: page.First(), page.Last()?

Comment (JuanCarlosGI) [reply Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs]: Lance pointed out that Last() isn't optimized for List, so accessing through index is faster/better

**Pull request merged**